sale_blanket_order/migrations/14.0.1.2.2/pre-migrate.py

```diff
@@ -0,0 +1,19 @@
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    _logger.info("Updating states for sale_blanket_order " + str(version))
+
+    cr.execute(
+        """
+        UPDATE sale_blanket_order sbo
+        SET state = 'cancel'
+        WHERE sbo.state = 'expired'
+        """
+    )
```

Comment:
**nitpick(non-blocking):** I personally think that `openupgrade.logged_query()` fits better here :)